toolchain/lower/file_context.h

```diff
@@ -157,6 +188,50 @@ class FileContext {
 
   auto BuildVtable(const SemIR::Class& class_info) -> llvm::GlobalVariable*;
 
+  // Function is used to track which specifics were lowered for each generic.
+  // These are added one by one while lowering their definitions.
+  auto AddLoweredSpecificForGeneric(SemIR::GenericId generic_id,
+                                    SemIR::SpecificId specific_id) {
+    lowered_specifics_[generic_id.index].push_back(specific_id);
+  }
+  // Initialize a return a SpecificFunctionFingerprint* instance for specific.
+  // The internal of the fingerprint are populated during and after lowering
+  // the function body of that specific.
+  auto InitializeFingerprintForSpecific(SemIR::SpecificId specific_id)
+      -> SpecificFunctionFingerprint* {
+    if (!specific_id.has_value()) {
+      return nullptr;
+    }
+    return &lowered_specific_fingerprint_[specific_id.index];
+  }
+
+  // Entry point for coalescing equivalent specifics.
```

Comment:
Would this be a good spot to explain what coalescing equivalent specifics means and involves, at least at a high level?